sources/web/datalab/polymer/modules/GapiManager.ts

```diff
@@ -114,50 +143,61 @@ class GapiManager {
       projectId,
       tableId,
     };
-    return GapiManager._loadBigQuery().then(() => gapi.client.bigquery.tables.get(request));
+    return GapiManager._loadBigQuery()
+      .then(() => gapi.client.bigquery.tables.get(request));
   }
 
-  private static _initClient(signInChangedCallback: (signedIn: boolean) => void) {
-    // Initialize the client with API key and People API, and initialize OAuth with an
-    // OAuth 2.0 client ID and scopes (space delimited string) to request access.
-    gapi.client.init({
+  public static listenForSignInChanges(signInChangedCallback: (signedIn: boolean) => void):
+      Promise<void> {
+    return this.loadGapi()
+      .then(() => {
+        // If auth status is already know, initialize the callback now
+        if (gapi && gapi.auth2) {
+          const signedIn = gapi.auth2.getAuthInstance().isSignedIn.get();
+          signInChangedCallback(signedIn);
+        }
+
+        // Listen for auth changes
+        gapi.auth2.getAuthInstance().isSignedIn.listen(() => {
```

Comment:
If line 155 is valid, then this line doesn't look right. If the condition on line 155 returns false, then this line will fail because gapi.auth2 is not defined.